source/h1_connection.c

```diff
@@ -1261,6 +1280,15 @@ static int s_decoder_on_done(void *user_data) {
     if (err) {
         return AWS_OP_ERR;
     }
+    if (incoming_stream->base.on_metrics) {
+        aws_high_res_clock_get_ticks(&incoming_stream->base.metrics.receive_end_timestamp_ns);
+        AWS_ASSERT(incoming_stream->base.metrics.receive_start_timestamp_ns != 0);
+        AWS_ASSERT(
+            incoming_stream->base.metrics.receive_end_timestamp_ns >=
+            incoming_stream->base.metrics.receive_start_timestamp_ns);
+        incoming_stream->base.metrics.receiving_duration_ns = incoming_stream->base.metrics.receive_end_timestamp_ns -
+                                                              incoming_stream->base.metrics.receive_start_timestamp_ns;
+    }
```

Comment:
We should move this block after 
```
    /* Otherwise the incoming stream is finished decoding and we will update it if needed */
    incoming_stream->is_incoming_message_done = true;
```
this won't be correct if we recieve an informational response.